packages/protocol/test/backward/version.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  ContractVersion,
+  ContractVersionDelta,
+  Delta,
+  DeltaUtil,
+} from '@celo/protocol/lib/backward/version'
+import { assert } from 'chai'
+
+describe('#version()', () => {
+  describe('ContractVersion', () => {
+    it('isValid fails on invalid string', () => {
+      assert.isFalse(ContractVersion.isValid('1.2.3'))
+      assert.isFalse(ContractVersion.isValid('1.2'))
+      assert.isFalse(ContractVersion.isValid('1.2.3.5.'))
+      assert.isFalse(ContractVersion.isValid('test'))
+      assert.isFalse(ContractVersion.isValid('1.a.3.4'))
+    })
+    it('isValid true on valid string', () => {
+      assert.isTrue(ContractVersion.isValid('1.2.3.4'))
+      assert.isTrue(ContractVersion.isValid('-1.0.9.787'))
+      assert.isTrue(ContractVersion.isValid('10.0.224.9'))
+    })
+  })
+  describe('with a new ContractVersion instance', () => {
+    it('properties are well set', () => {
+      const version = new ContractVersion(1, 2, 3, 4)
+      assert.equal(version.storage, 1)
+      assert.equal(version.major, 2)
+      assert.equal(version.minor, 3)
+      assert.equal(version.patch, 4)
+    })
+    it('toString works', () => {
+      const version = new ContractVersion(2, 7, 9, 1)
+      assert.equal(version.toString(), '2.7.9.1')
+    })
+  })
+
+  describe('ContractVersionDelta.fromChanges', () => {
+    it('sets properly the storage Delta', () => {
+      const cvd = ContractVersionDelta.fromChanges(true, false, true, true)
+      assert.equal(cvd.storage, Delta.Increment)
+      assert.equal(cvd.major, Delta.Reset)
+      assert.equal(cvd.minor, Delta.Reset)
+      assert.equal(cvd.patch, Delta.Reset)
+
+      const cvd2 = ContractVersionDelta.fromChanges(false, false, true, true)
```

Comment:
This feels like a new `it` test case since it doesn't use any of the previous setup.